src/Processors/Formats/Impl/CHColumnToArrowColumn.cpp

```diff
@@ -235,27 +235,30 @@ namespace DB
     }
 
     template<typename T>
-    static PaddedPODArray<Int64> extractIndexesImpl(ColumnPtr column, size_t start, size_t end)
+    static PaddedPODArray<Int64> extractIndexesImpl(ColumnPtr column, size_t start, size_t end, bool shift)
     {
         const PaddedPODArray<T> & data = assert_cast<const ColumnVector<T> *>(column.get())->getData();
         PaddedPODArray<Int64> result;
         result.reserve(end - start);
-        std::transform(data.begin() + start, data.begin() + end, std::back_inserter(result), [](T value) { return Int64(value); });
+        if (shift)
+            std::transform(data.begin() + start, data.begin() + end, std::back_inserter(result), [](T value) { return Int64(value) - 1; });
```

Comment:
Cosmetic: Surely a matter of personal taste, but maybe

```
std::transform(data.begin() + start, data.begin() + end, std::back_inserter(result), [shift](T value) { return Int64(value) - shift; });
```
instead of l. 243-246?

Reply:
Wouldn't it be less effective in case when shift is false? We will perform redundunt extra operation on each row (but maybe compiler will optimize it, I am not sure).

Reply:
Yes, calculating the shift for each value is theoretically more expensive than doing it once. But I doubt there is any practical difference at all, even if the compiler would not optimize it. The cost of memcpy-ing data from/to Arrow is larger than spending a few cycles on extra calculations.

But it is really up to you, I see it as a cosmetic issue 😄